moto/awslambda/models.py

```diff
@@ -1813,9 +1883,7 @@ def create_event_source_mapping(self, spec: Dict[str, Any]) -> EventSourceMappin
                 raise RESTError("InvalidParameterValueException", f"Missing {param}")
 
         # Validate function name
-        func = self._lambdas.get_function_by_name_or_arn(spec.get("FunctionName", ""))
-        if not func:
-            raise RESTError("ResourceNotFoundException", "Invalid FunctionName")
```

Comment:
The exception is now thrown within `get_function_by_name_or_arn_with_qualifier`, `func` is never `None` here